test/unit_test_input_parser.c

```diff
@@ -1,6 +1,166 @@
 #include "unit_test_input_parser.h"
 
+// WARNING !!! This test may seg fault if it fails!
```

Comment:
Does it have to segfault on failure?
Can't it check for those conditions, and just fail before segfaulting?